src/ORM/Search/SearchContext.php

```diff
@@ -104,7 +105,18 @@ public function __construct($modelClass, $fields = null, $filters = null)
      */
     public function getSearchFields()
     {
-        return ($this->fields) ? $this->fields : singleton($this->modelClass)->scaffoldSearchFields();
+        if ($this->fields !== null) {
+            return $this->fields;
+        }
+
+        $singleton = singleton($this->modelClass);
+        if (!$singleton->hasMethod('scaffoldSearchFields')) {
+            throw new LogicException(
+                'Cannot dynamically determine search fields. Pass the fields to setFields()'
+                . " or implement a scaffoldSearchFields() method on {$this->modelClass}"
+            );
+        }
+        return $singleton->scaffoldSearchFields();
```

Comment:
This is technically a bugfix, allowing classes to implement a `scaffoldSearchFields` method instead of passing fields to the `SearchContext` instance directly.